parachain/runtime/tests/test_nomination.rs

```diff
@@ -0,0 +1,354 @@
+        disable_nomination();
+        assert_ok!(deregister_operator(VAULT));
+    });
+}
+
+#[test]
+fn integration_test_operators_with_nonzero_nomination_can_force_opt_opt() {
+    test_with_nomination_enabled_and_operator_registered(|| {
+        assert_nominate_collateral(USER, VAULT, DEFAULT_NOMINATION);
+        let user_collateral_balance_before_force_refund = CollateralPallet::get_balance_from_account(&account_of(USER));
+        assert_eq!(user_collateral_balance_before_force_refund, 900000);
+        assert_ok!(deregister_operator(VAULT));
+        let user_collateral_balance_after_force_refund = CollateralPallet::get_balance_from_account(&account_of(USER));
+        assert_eq!(user_collateral_balance_after_force_refund, 900000 + DEFAULT_NOMINATION);
+    });
+}
+
+#[test]
+fn integration_test_non_operators_cannot_have_collateral_nominated() {
+    test_with_nomination_enabled(|| {
+        assert_noop!(
+            Call::Nomination(NominationCall::deposit_nominated_collateral(
+                account_of(VAULT),
+                DEFAULT_BACKING_COLLATERAL
+            ))
+            .dispatch(origin_of(account_of(USER))),
+            NominationError::VaultNotOptedInToNomination
+        );
+    });
+}
+
+#[test]
+fn integration_test_operators_can_have_collateral_nominated() {
+    test_with_nomination_enabled_and_operator_registered(|| {
+        assert_nominate_collateral(USER, VAULT, DEFAULT_NOMINATION);
+        let nominator_collateral = get_nominator_collateral();
+        assert_eq!(nominator_collateral, DEFAULT_NOMINATION);
+        assert_total_nominated_collateral_is(VAULT, DEFAULT_NOMINATION);
+    });
+}
+
+#[test]
+fn integration_test_operators_cannot_withdraw_nominated_collateral() {
+    test_with_nomination_enabled_and_operator_registered(|| {
+        assert_nominate_collateral(USER, VAULT, DEFAULT_NOMINATION);
+        assert_noop!(
+            request_operator_collateral_withdrawal(VAULT, DEFAULT_BACKING_COLLATERAL + 1),
+            NominationError::InsufficientCollateral
+        );
+    });
+}
+
+#[test]
+fn integration_test_nominated_collateral_cannot_exceed_max_nomination_ratio() {
+    test_with_nomination_enabled_and_operator_registered(|| {
+        assert_noop!(
+            nominate_collateral(USER, VAULT, DEFAULT_BACKING_COLLATERAL),
+            NominationError::DepositViolatesMaxNominationRatio
+        );
+    });
+}
+
+#[test]
+fn integration_test_nominated_collateral_prevents_replace_requests() {
+    test_with_nomination_enabled_and_operator_registered(|| {
+        assert_nominate_collateral(USER, VAULT, DEFAULT_NOMINATION);
+        assert_noop!(
+            Call::Replace(ReplaceCall::request_replace(0, DEFAULT_BACKING_COLLATERAL))
+                .dispatch(origin_of(account_of(VAULT))),
+            ReplaceError::VaultUsesNominatedCollateral
+        );
+    });
+}
+
+#[test]
+fn integration_test_operators_with_zero_nomination_can_request_replacement() {
```

Comment:
What happens when after requesting a replace, it gets some nomination, and then a vault does `accept_replace`?

Reply:
Hm, I wasn't completely aware of the `replace` flow. In this case replace requests should be forbidden altogether